sdks/java/core/src/main/java/org/apache/beam/sdk/testing/BigqueryMatcher.java

```diff
@@ -185,11 +186,13 @@ private Credential getDefaultCredential(HttpTransport transport, JsonFactory jso
   private String generateHash(List<TableRow> rows) {
     List<HashCode> rowHashes = Lists.newArrayList();
     for (TableRow row : rows) {
-      List<HashCode> cellHashes = Lists.newArrayList();
+      List<String> cellsInOneRow = Lists.newArrayList();
       for (TableCell cell : row.getF()) {
-        cellHashes.add(Hashing.sha1().hashString(cell.toString(), StandardCharsets.UTF_8));
+        cellsInOneRow.add(Objects.toString(cell.getV()));
```

Comment:
Objects.toString() just calls the object's toString method and then returns "null" if toString is null -- can you guarantee that every type that can be returned from a BigQuery cell has a stable toString method? Otherwise this doesn't quite solve the problem.